### ISSUE TITLE
Add singularity-grpc client

### ISSUE THREAD

**Issue by goord** [eWaterCycle/grpc4bmi#1]
...but should it run in a child process?

**Comment by goord** [comment]
Use [spython](https://pypi.org/project/spython/) for interacting with such containers:.

**Comment by sverhoeven** [comment]
Able to start a singularity container (v3.0.3) with
```bash
# put config file inside RhineMeuse30min/ dir
mkdir output
singularity pull docker://ewatercycle/pcrg-grpc4bmi:latest
singularity run -B RhineMeuse30min:/data/input -B output:/data/output pcrg-grpc4bmi-latest
```
and connect to it on port 55555 and run the model with
```python
model = BmiClient(grpc.insecure_channel("localhost:55555"))
model.initialize('setup_natural_test.ini')
```

Non default port fails
```bash
# Below still runs on 55555
SINGULARITYENV_BMI_PORT=55566 singularity run -B RhineMeuse30min:/data/input -B output:/data/output pcrg-grpc4bmi-latest
SINGULARITYENV_BMI_PORT=55566 singularity exec pcrg-grpc4bmi-latest env |grep BMI_PORT
# outputs BMI_PORT=55555
```

**Comment by sverhoeven** [comment]
Workarounds for non-default port 

Below runs on 55566, but requires image unique command
```bash
singularity exec -B RhineMeuse30min:/data/input -B output:/data/output pcrg-grpc4bmi_latest.sif run-bmi-server --path /opt/PCR-GLOBWB_model/model --port 55566
```
Below runs on 55566, but requires root
```bash
sudo singularity run --net --network-args portmap=55566:55555/tcp -B RhineMeuse30min:/data/input -B output:/data/output pcrg-grpc4bmi_latest.sif
```

Def file with port fix, requires root to build, but runs on overwritten port
pcrg-grpc4bmi_sing.def:
```
Bootstrap: docker
From: ewatercycle/pcrg-grpc4bmi:latest

%post
sed -i 's/"55555"/${BMI_PORT:-55555}/' /.singularity.d/env/10-docker2singularity.sh
```
```bash
sudo singularity build pcrg-grpc4bmi_sing.sif pcrg-grpc4bmi_sing.def
SINGULARITYENV_BMI_PORT=55566 singularity run -B RhineMeuse30min:/data/input -B output:/data/output pcrg-grpc4bmi_sing.sif
```

**Comment by sverhoeven** [comment]
To run isolated use
```bash
singularity exec --containall --cleanenv --pwd /data/input \
-B RhineMeuse30min:/data/input -B output:/data/output \
docker://ewatercycle/pcrg-grpc4bmi:latest \
run-bmi-server --path /opt/PCR-GLOBWB_model/model --port 55566
```
This will build a singularity image in `~/.singularity/cache` the first time.

If the `--path` can be set using BMI_PATH env var then the command would be more re-usable.

**Comment by sverhoeven** [comment]
More people unable to overwrite Docker envs, see https://github.com/sylabs/singularity/issues/2668 and https://github.com/sylabs/singularity/issues/2451

https://github.com/sylabs/singularity/pull/2669 should have fixed but does not work with
```bash
BMI_PORT=55566 singularity exec docker://ewatercycle/pcrg-grpc4bmi:latest env |grep BMI_PORT
# or
SINGULARITYENV_BMI_PORT=55566 singularity exec docker://ewatercycle/pcrg-grpc4bmi:latest env |grep BMI_PORT
```
Still reports port 55555, from the [Dockerfile](https://github.com/eWaterCycle/grpc4bmi-examples/blob/master/pcrglob/Dockerfile)

**Comment by goord** [comment]
Fixed by #31
